### PR TITLE
support more tensor init function

### PR DESCRIPTION
**Pull request by QiJune**
This PR also change `int` in C++ header/source file to `int64_t`.

### CODE REVIEW

File: cgotorch/cgotorch.cc [hunk 1]
```diff
@@ -22,14 +23,83 @@ char *exception_str(const std::exception &e) {
   return r;
 }
 
+std::unordered_map<std::string, torch::nn::init::FanModeType> fan_mode_map = {
+    {"fan_in", torch::kFanIn},
+    {"fan_out", torch::kFanOut},
+};
+
+std::unordered_map<std::string, torch::nn::init::NonlinearityType>
```
Comment (wangkuiyi) [inline cgotorch/cgotorch.cc:31]: It seems that this global variable is only used by `KaimingUniform_`. Should we move it into `KaimingUniform_` as a local static variable?

File: cgotorch/cgotorch.cc [hunk 2]
```diff
@@ -22,14 +23,83 @@ char *exception_str(const std::exception &e) {
   return r;
 }
 
+std::unordered_map<std::string, torch::nn::init::FanModeType> fan_mode_map = {
```
Comment (wangkuiyi) [inline cgotorch/cgotorch.cc:26]: This global variable is only used by `KaimingUniform_` as `non_linearity_map`.

File: cgotorch/cgotorch.h [hunk 3]
```diff
@@ -10,7 +10,20 @@ extern "C" {
 
 typedef void *Tensor;
 typedef void *Optimizer;
-char *RandN(int rows, int cols, int require_grad, Tensor *result);
+// torch.randn
+char *RandN(int64_t *size, int64_t length, int64_t require_grad,
```
Comment (wangkuiyi) [inline cgotorch/cgotorch.h:14]: Should we make the return type from `char *` to `const char *`?

File: cgotorch/cgotorch.cc (hunk 1 above)
Comment (QiJune) [reply cgotorch/cgotorch.cc:31]: There is another parameter initialization named [`kaiming_normal_`](https://pytorch.org/cppdocs/api/function_namespacetorch_1_1nn_1_1init_1ac8a913c051976a3f41f20df7d6126e57.html).  It also uses these two maps. So let's take it as global variables.

File: cgotorch/cgotorch.h (hunk 3 above)
Comment (QiJune) [reply cgotorch/cgotorch.h]: Yes, we do not modify the return error message. `const char*` is more accurate.

**Pull request merged**